lib/datadog/core/remote/client.rb

```diff
@@ -4,14 +4,45 @@
 
 require_relative 'configuration'
 require_relative 'dispatcher'
-require_relative '../../appsec/processor/rule_merger'
-require_relative '../../appsec/processor/rule_loader'
 
 module Datadog
   module Core
     module Remote
       # Client communicates with the agent and sync remote configuration
       class Client
+        class SyncError < StandardError; end
+
+        class << self
+          attr_reader :products, :capabilities, :receivers
+
+          def init
+            @capabilities = []
+            @products = []
+            @receivers = []
+          end
+
+          def register_capabilities(capabilities)
+            @capabilities.concat(capabilities)
+          end
+
+          def register_receivers(receivers)
+            @receivers.concat(receivers)
+          end
+
+          def register_products(products)
+            @products.concat(products)
+          end
+
+          def binary_capabilities
+            return @binary_capabilities if defined?(@binary_capabilities)
```

Comment:
I read recently that `defined?` can be slow.

Better use `@binary_capabilities.nil?`, and set `@binary_capabilities = nil` in `initialize`.